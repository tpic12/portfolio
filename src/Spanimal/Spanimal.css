.links a {
  text-decoration: none;
  color: rgb(62, 72, 117);
}

.links a:hover {
  font-weight: 700;
  color: rgb(189, 89, 52);
}

.spanimal-section {
  width: 100vw;
  min-height: 100vh;
  background-color: white;
  color: rgb(62, 72, 117);
  display: flex;
  flex-direction: column;
  /* margin-top: 100px; */
  justify-content: flex-start;
  align-items: center;
  padding: 0 15px;
}

/* .spanimal-section h2 {
  padding-top: 100px;
  margin-bottom: 0;
} */

.links p {
  padding: 0;
}

.spanimals-logo-arrows {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  width: 100%;
  align-items: center;
  padding-top: 60px;
}

.spanimals-logo-arrows a {
  text-decoration: none;
  color: rgb(62, 72, 117);
}

.spanimal-circle {
  width: 100vw;
  height: 375px;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  border: 7px solid #ededed;
  border-radius: 200px;
  margin: 0 10px;
  background-color: rgb(62, 72, 117);
  color: white;
  text-align: center;
  padding: 10px 30px;
}

.spanimal-circle button {
  background-color: transparent;
  border: none;
  cursor: pointer;
}

.spanimal-circle button:hover {
  box-shadow: 0px 0px 10px #ededed;
  border-radius: 10px;
}

.spanimal-circle img {
  width: 100px;
  animation: pulse 1s linear;
  animation-iteration-count: infinite;
}

.tech-logos {
  width: 50%;
  margin-top: 10px;
  display: flex;
  justify-content: space-evenly;
}

@media screen and (min-width: 500px) {
  .spanimal-circle {
    font-size: 20px;
  }

  .spanimal-section h2 {
    font-size: 30px;
  }
}

@media screen and (min-width: 720px) {
  .spanimal-circle {
    max-width: 900px;
    font-size: 24px;
  }
}