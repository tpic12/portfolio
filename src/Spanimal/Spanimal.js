import React, { Component } from "react";
import Slides from "../Slides/slides";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import Logo from "../Images/spanimal-logo.png";
import Spanimals0 from "../Images/spanimals0.png";
import Spanimals1 from "../Images/spanimals1.png";
import Spanimals2 from "../Images/spanimals2.png";
import "./Spanimal.css";

class Spanimal extends Component {
  state = {
    slideshow: false,
    imageArr: [Spanimals0, Spanimals1, Spanimals2],
  };

  handlesSlideshow = () => {
    const { slideshow } = this.state;
    this.setState({ slideshow: !this.state.slideshow });
    document.documentElement.style.overflow = !slideshow ? "hidden" : "scroll";
    document.body.scroll = !slideshow;
  };
  render() {
    const { slideshow } = this.state;
    return (
      <section id="spanimals-section" className="spanimal-section snap">
        {slideshow && (
          <Slides
            slideshow={this.handlesSlideshow}
            image={this.state.imageArr}
          />
        )}
        <h2>Spanimals</h2>
        <div className="links">
          <p>
            {" "}
            <a
              href="https://spanimal.now.sh/"
              target="_blank"
              rel="noopener noreferrer"
            >
              Live Link
            </a>{" "}
            |{" "}
            <a
              href="https://github.com/tpic12/spanimal-client"
              target="_blank"
              rel="noopener noreferrer"
            >
              GitHub
            </a>
          </p>
        </div>
        <div className="spanimal-circle">
          <div className="spanimals-logo-arrows">
            <a href="#linus-section">
              <FontAwesomeIcon icon="chevron-left" size="lg" />
            </a>
            <button onClick={this.handlesSlideshow}>
              {!slideshow && <img src={Logo} alt="spanimals logo" />}
            </button>
            <div></div>
          </div>
          <p>
            A Learning Language platform that teaches the Spanish translations
            for animals using the spaced-repetition technique. Users can
            sign-up, start learning and keep track of their scores per word, and
            across the language.
          </p>
        </div>
        <div className="tech-logos">
          <FontAwesomeIcon icon={["fab", "react"]} size="2x" />
          <FontAwesomeIcon icon={["fab", "css3"]} size="2x" />
          <FontAwesomeIcon icon={["fab", "node-js"]} size="2x" />
          <FontAwesomeIcon icon={["fas", "database"]} size="2x" />
        </div>
        <div className="extra-tech">
          <ul>
            <li>Express</li>
            <li>Mocha</li>
            <li>Chai</li>
            <li>Cypress</li>
          </ul>
        </div>
      </section>
    );
  }
}

export default Spanimal;
/**
 *         <div className="spanimal-text">
          <h2>Spanimal</h2>
          <p>
            A Learning Language platform that teaches the Spanish translations
            for animals using the spaced-repetition technique. Users can
            sign-up, start learning and keep track of their scores per word, and
            across the language.
          </p>
        </div>
        <div className="links">
          <p>
            {" "}
            <a
              href="https://spanimal.now.sh/"
              target="_blank"
              rel="noopener noreferrer"
            >
              Live Link
            </a>{" "}
            |{" "}
            <a
              href="https://github.com/tpic12/spanimal-client"
              target="_blank"
              rel="noopener noreferrer"
            >
              GitHub
            </a>
          </p>
        </div>
        <div className="tech-stack">
          <h3>Technology Used:</h3>
          <ul className="tech-stack-list">
            <li>React</li>
            <li>Express</li>
            <li>Javascript</li>
            <li>Node</li>
            <li>PostgreSQL</li>
            <li>JWT</li>
            <li>Cypress</li>
          </ul>
        </div>
        <div className="screenshots">
          <p>screenshots here</p>
          <p>screenshots here</p>
          <p>screenshots here</p>
        </div>
 */
