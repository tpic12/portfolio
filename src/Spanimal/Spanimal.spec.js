import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Spanimal from "./Spanimal";

describe("<Landing />", () => {
  it("renders by default", () => {
    const wrapper = shallow(<Spanimal />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
