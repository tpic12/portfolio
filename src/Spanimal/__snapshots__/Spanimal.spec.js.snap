// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> renders by default 1`] = `
<section
  className="spanimal-section snap"
  id="spanimals-section"
>
  <h2>
    Spanimals
  </h2>
  <div
    className="links"
  >
    <p>
       
      <a
        href="https://spanimal.now.sh/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Live Link
      </a>
       
      |
       
      <a
        href="https://github.com/tpic12/spanimal-client"
        rel="noopener noreferrer"
        target="_blank"
      >
        GitHub
      </a>
    </p>
  </div>
  <div
    className="spanimal-circle"
  >
    <div
      className="spanimals-logo-arrows"
    >
      <a
        href="#linus-section"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="chevron-left"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="lg"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </a>
      <button
        onClick={[Function]}
      >
        <img
          alt="spanimals logo"
          src="spanimal-logo.png"
        />
      </button>
      <div />
    </div>
    <p>
      A Learning Language platform that teaches the Spanish translations for animals using the spaced-repetition technique. Users can sign-up, start learning and keep track of their scores per word, and across the language.
    </p>
  </div>
  <div
    className="tech-logos"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fab",
          "react",
        ]
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="2x"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fab",
          "css3",
        ]
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="2x"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fab",
          "node-js",
        ]
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="2x"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fas",
          "database",
        ]
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="2x"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </div>
  <div
    className="extra-tech"
  >
    <ul>
      <li>
        Express
      </li>
      <li>
        Mocha
      </li>
      <li>
        Chai
      </li>
      <li>
        Cypress
      </li>
    </ul>
  </div>
</section>
`;
